Add support for Kind OVNK IPV6 deployment

yboaron opened the pull request:
Depends on: https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4955

yboaron committed:
Add support for Kind OVNK IPV6 deployment

This PR depends on ovn-kubernetes PR [1]

[1]
https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4955

Signed-off-by: Yossi Boaron <yboaron@redhat.com>

yboaron commented:
@mkolesnik , QQ: I'm getting [1] error when trying to deploy Kind clusters with OVNK and IPV6 with this PR(with **make clusters using=ipv6-stack,ovn**), while running directly **./ovn-kubernetes/contrib/kind.sh ....**  things looks fine, any idea ?

[1] 
`[cluster2] ++ sysctl net.ipv6.conf.all.forwarding
[cluster1] ++ sysctl net.ipv6.conf.all.forwarding
[cluster2] ++ awk '{print $3}'
[cluster1] ++ awk '{print $3}'
[cluster2] + TMPVAR=0
[cluster2] + echo 'net.ipv6.conf.all.forwarding is equal to 0'
[cluster2] net.ipv6.conf.all.forwarding is equal to 0
[cluster2] + '[' 0 '!=' 1 ']'
[cluster1] + TMPVAR=0
[cluster2] + '[' true == true ']'
[cluster2] + sudo sysctl -w net.ipv6.conf.all.forwarding=1
[cluster1] + echo 'net.ipv6.conf.all.forwarding is equal to 0'
[cluster1] net.ipv6.conf.all.forwarding is equal to 0
[cluster1] + '[' 0 '!=' 1 ']'
[cluster1] + '[' true == true ']'
[cluster1] + sudo sysctl -w net.ipv6.conf.all.forwarding=1
[cluster2] sysctl: permission denied on key "net.ipv6.conf.all.forwarding"
[cluster1] sysctl: permission denied on key "net.ipv6.conf.all.forwarding"
[cluster2] Failed to run './ovn-kubernetes/contrib/kind.sh -ov localhost:5000/ovn-daemonset-f:latest -cn cluster2 -ric -n4 -i6 -sw -lr `

yboaron commented on package/Dockerfile.shipyard-dapper-base:
```diff
@@ -4,7 +4,7 @@ FROM fedora:41
 ARG UPX_LEVEL=-5
 ENV DAPPER_HOST_ARCH=amd64 SHIPYARD_DIR=/opt/shipyard SHELL=/bin/bash \
-    DAPPER_RUN_ARGS="--net=kind"
+    DAPPER_RUN_ARGS="--net=kind --cap-add=NET_ADMIN --privileged=true"
```
We only need the extra networking permissions to container for OVN/IPV6 deployment.

@mkolesnik @skitt , is it possible to set DAPPER_RUN_ARGS content based on CNI, IP family env vars ?

skitt replied:
`DAPPER_RUN_ARGS` is a bit of a cheat — it’s not used as an environment variable as such inside the container, so nothing running in the container can set it meaningfully. The `.dapper` scripts extracts the value of the variable from the container image before running it.

However since we decide what `.dapper` does, anything can be done. Could you explain what the exacts criteria (as in, which values in which files) are and which containers need to run with privileges?

yboaron replied:
For IPV6 OVN deployment (**make clusters using=ipv6-stack,ovn**) OVN-K script [1] sets net.ipv6.conf.all.forwarding and sudo sysctl -w net.ipv6.conf.all.disable_ipv6 .
so, container should run with net privileges.

[1] 
https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/contrib/kind.sh#L642

skitt replied:
It only does so if the values don’t match what it expects. Unfortunately [GH doesn’t enable IPv6 forwarding](https://github.com/submariner-io/shipyard/actions/runs/12792227597/job/35662173778?pr=1813) by default, so changing that would be necessary.

However it would be possible to do this as an extra step before running the jobs, instead of making the containers themselves privileged…

skitt replied:
See https://github.com/submariner-io/shipyard/actions/runs/12792280698/job/35662349880?pr=1813

yboaron replied:
> It only does so if the values don’t match what it expects

Yep, without net privileges we can't even read kernel's ipv6 config (sysctl net.ipv6.conf.all.forwarding)

> However it would be possible to do this as an extra step before running the jobs, instead of making the containers themselves privileged…

Maybe  I'm missing something here, this idea will be useful only on GH,  will it be possible to deploy Kind+IPV6,OVN locally without making the containers privileged?

skitt replied:
> Yep, without net privileges we can't even read kernel's ipv6 config (sysctl net.ipv6.conf.all.forwarding)

That’s surprising, it works for me and should work fine as long as `/proc/sys` is mounted in the container (`sysctl` literally only reads from that).

skitt replied:
For local deployments, we can check the settings early and give an appropriate error, same as for other `sysctl` settings. In most cases I expect developer machines will be set up with IPv6 enabled and forwarded anyway.

yboaron committed:
Workaround until ipv6 bugfix merge

Workaround until commit [1] is delivered in ovn-kubernetes branch

[1]
https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4955

Signed-off-by: Yossi Boaron <yboaron@redhat.com>

skitt commented:
Would it be possible to add an IPv6 deployment in the deployment tests, to check that this does what is expected?
